### ISSUE TITLE
Strongloop installation error on windows Server 2012 R2 Std

### ISSUE THREAD

**Issue by jaideepdas** [strongloop/loopback#1685]
I am getting the following error while installing strongloop on windows server 2012 R2
```
C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\V140\Microsoft.CppBuild.targets(312,5): error MSB3491: Could not writ
e lines to file "Release\obj\validation\validation.tlog\validation.lastbuildstate". The specified path, file name, or b
oth are too long. The fully qualified file name must be less than 260 characters, and the directory name must be less t
han 248 characters. [C:\Users\Administrator\AppData\Roaming\npm\node_modules\strongloop\node_modules\strong-arc\node_mo
dules\strong-mesh-client\node_modules\engine.io\node_modules\ws\node_modules\utf-8-validate\build\validation.vcxproj]
gyp ERR! build error
gyp ERR! stack Error: `C:\Program Files (x86)\MSBuild\14.0\bin\msbuild.exe` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onExit (D:\apps\nodejs\node_modules\npm\node_modules\node-gyp\lib\build.js:270:23)
gyp ERR! stack     at emitTwo (events.js:87:13)
gyp ERR! stack     at ChildProcess.emit (events.js:172:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)
ERR! System Windows_NT 6.3.9600
gyp ERR! command "D:\\apps\\nodejs\\node.exe" "D:\\apps\\nodejs\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gy
p.js" "rebuild"
gyp ERR! cwd C:\Users\Administrator\AppData\Roaming\npm\node_modules\strongloop\node_modules\strong-arc\node_modules\str
ong-mesh-client\node_modules\engine.io\node_modules\ws\node_modules\utf-8-validate
gyp ERR! node -v v4.1.0
gyp ERR! node-gyp -v v3.0.3
gyp ERR! not ok
npm WARN optional dep failed, continuing utf-8-validate@1.2.1
```
Similar errors keeps popping up and then it completely dies out. I suspect its a IA64 vs AMD64 issue as it works fine on my amd64 bit process with windows 10

**Comment by jaideepdas** [comment]
some more error tracebacks
```
LINK : fatal error LNK1194: cannot delay-load 'node.exe' due to import of data symbol '"__declspec(dllimport) const v8:
:ArrayBuffer::Allocator::`vftable'" (__imp_??_7Allocator@ArrayBuffer@v8@@6B@)'; link without /DELAYLOAD:node.exe [D:\ap
ps\nodejs\node_modules\strongloop\node_modules\strong-debugger\build\debugger.vcxproj]
gyp ERR! build error
gyp ERR! stack Error: `C:\Program Files (x86)\MSBuild\14.0\bin\msbuild.exe` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onExit (D:\apps\nodejs\node_modules\npm\node_modules\node-gyp\lib\build.js:270:23)
gyp ERR! stack     at emitTwo (events.js:87:13)
gyp ERR! stack     at ChildProcess.emit (events.js:172:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)
gyp ERR! System Windows_NT 6.3.9600
gyp ERR! command "D:\\apps\\nodejs\\node.exe" "D:\\apps\\nodejs\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gy
p.js" "rebuild"
gyp ERR! cwd D:\apps\nodejs\node_modules\strongloop\node_modules\strong-debugger
gyp ERR! node -v v4.1.0
gyp ERR! node-gyp -v v3.0.3
gyp ERR! not ok
D:\apps\nodejs
└── (empty)

npm ERR! Windows_NT 6.3.9600
npm ERR! argv "D:\\apps\\nodejs\\node.exe" "D:\\apps\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "install" "-g" "strong
loop"
npm ERR! node v4.1.0
npm ERR! npm  v3.3.4
npm ERR! code ELIFECYCLE

npm ERR! strong-debugger@1.0.0 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the strong-debugger@1.0.0 install script 'node-gyp rebuild'.
npm ERR! This is most likely a problem with the strong-debugger package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get their info via:
npm ERR!     npm owner ls strong-debugger
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     D:\data\ssgfluux\npm-debug.log
npm ERR! code 1
```

**Comment by morgdenn** [comment]
I'm having the same problem, except I'm using Win 10.

LINK : fatal error LNK1194: cannot delay-load 'node.exe' due to import of data symbol '"__declspec(dllimport) const v8::ArrayBuffer::Allocator::`vftable'" (__imp_??_7Allocator@ArrayBuffer@v8@@6B@)'; link without /DELAYLOAD:node.exe [C:\Program Files\nodejs\node_modules\strong-debugger\build\debugger.vcxproj]
gyp ERR! build error
gyp ERR! stack Error: `C:\Program Files (x86)\MSBuild\14.0\bin\msbuild.exe` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onExit (C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\lib\build.js:270:23)
gyp ERR! stack     at emitTwo (events.js:87:13)
gyp ERR! stack     at ChildProcess.emit (events.js:172:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)
gyp ERR! System Windows_NT 10.0.10240
gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd C:\Program Files\nodejs\node_modules\strong-debugger
gyp ERR! node -v v4.1.0
gyp ERR! node-gyp -v v3.0.1
gyp ERR! not ok
npm ERR! Windows_NT 10.0.10240
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "install" "-g" "strong-debugger" "--msvs_version=2015"
npm ERR! node v4.1.0
npm ERR! npm  v3.3.3
npm ERR! code ELIFECYCLE

npm ERR! strong-debugger@1.0.0 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the strong-debugger@1.0.0 install script 'node-gyp rebuild'.
npm ERR! This is most likely a problem with the strong-debugger package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get their info via:
npm ERR!     npm owner ls strong-debugger
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     C:\www\npm-debug.log

**Comment by jaideepdas** [comment]
@morgdenn are you also on xeon processor.

**Comment by superkhau** [comment]
@rmg ^

**Comment by rmg** [comment]
This is the same problem as strongloop/strongloop#253.

TL;DR: there's a bug in npm@3 (npm/npm#9643) that affects global modules with nested `optionalDependencies`. Unfortunately, that describes most of our globally installed modules.

For now the solution is `npm install -g npm@2`.

**Comment by rmg** [comment]
As an aside, `AMD64` == `x86_64` == `x86-64` == `x64` and applies to Xeon processes. `IA64`, on the other hand, refers to Itanium and to the best of my knowledge is completely unsupported by V8 and Node.

**Comment by superkhau** [comment]
See strongloop/strongloop#253 as recommended by @rmg. Closing.
